tests/refinements/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -2565,7 +2565,7 @@ let tests = [
     if (x.kind === null.toString()) {
     }
     // error, method on null
-    if ({ kind: 1 }.kind === null.toString()) {
+    if (({ kind: 1 }).kind === null.toString()) {
```

Comment:
This one is tricky. If something starts with a `{}` in a context where a `{}` is valid, then we need to add parenthesis.

- Context where {} is valid: arrow functions with direct result, `if/then/else/for/...` without `{}`
- Things that can output `{}` on the left position and not parenthesis: object expression on the left of binary expression (can be heavily nested), member expressions, conditionals...

So, that's a lot of cases to figure out and require going up the tree.

I think that those cases are rare enough that they are not worth the efforts trying to make it perfect. Having extra parenthesis here seems like a good trade-off.